### PR TITLE
 Make logger generate newline at EOL

### PR DESCRIPTION
**Pull request by Xaleth**

### CODE REVIEW

File: engine/source/src/core/utilities/Logger.cpp [hunk 1]
```diff
@@ -150,16 +150,16 @@ void Logger::logMessage(const std::string& errorFile, int lineNumber, LogVerbosi
 
 		// The "\r" prefix makes sure that if there is a new message, but it is the same, it will rewrite that line to the console, 
 		// rather than to print it out again and waste terminal space. Really useful, but may confuse people just as it did confuse me.
-		m_logFileHandle << '\r' << m_prevMessage << " (" << m_currentDuplicates << ")";
-		std::cout << '\r' << logMessageString << " (" << m_currentDuplicates << ")";
+		m_logFileHandle << '\r' << m_prevMessage << " (" << m_currentDuplicates << ")" << '\n';
```
Comment (beeperdeeper089) [inline engine/source/src/core/utilities/Logger.cpp]: These ``\n``s override the possible action of the carriage returns (``r``).

This is due to the carriage returns returning to the beginning of the *current* row, not to the last carriage return, thus, we should probably move out the ``\n``s on both.

Comment (Xaleth) [reply engine/source/src/core/utilities/Logger.cpp]: I don't understand a single word you said XD

Comment (beeperdeeper089) [reply engine/source/src/core/utilities/Logger.cpp]: welp, i need to work on explanations 😂

**Comment by beeperdeeper089** [comment]
Nice first contribution. 👍

**Pull request merged**